src/graph/sampler/metapath.cc

```diff
@@ -56,14 +65,27 @@ RandomWalkTracesPtr MetapathRandomWalk(
         if (succ.size() == 0)
           break;
         curr = succ[RandomEngine::ThreadLocal()->RandInt(succ.size())];
-        vertices.push_back(curr);
+        curr_vertices.push_back(curr);
         ++trace_length;
       }
 
-      trace_lengths.push_back(trace_length);
+      curr_trace_lengths.push_back(trace_length);
     }
 
-    trace_counts.push_back(curr_num_traces);
+    vertices_per_seed[seed_id] = curr_vertices;
+    trace_lengths_per_seed[seed_id] = curr_trace_lengths;
+    trace_counts[seed_id] = curr_num_traces;
+  }
+
+  for (uint64_t seed_id = 0; seed_id < num_seeds; ++seed_id) {
+    vertices.insert(
```

Comment:
Can we pre-alloc the vertices and trace_lengths?

Reply:
No.  The lengths of traces are not always the same.